File: ces-ddr-platform/ces-backend/app/auth/password.py
import asyncio

import bcrypt


class PasswordVerifier:
    DUMMY_PASSWORD_HASH = "$2a$10$7EqJtq98hPqEX7fNZaFWoOhiZCQGNuCIlnLzKeE/d1qFJKZ0f4uTe"

    async def verify(self, password: str, password_hash: str) -> bool:
        return await asyncio.to_thread(self.verify_sync, password, password_hash)

    async def hash(self, password: str) -> str:
        return await asyncio.to_thread(self.hash_sync, password)

    def verify_sync(self, password: str, password_hash: str) -> bool:
        try:
            return bcrypt.checkpw(password.encode(), password_hash.encode())
        except ValueError:
            return False

    def hash_sync(self, password: str) -> str:
        return bcrypt.hashpw(password.encode(), bcrypt.gensalt()).decode()

    def dummy_hash(self) -> str:
        return self.DUMMY_PASSWORD_HASH

File: ces-ddr-platform/ces-backend-go/internal/auth/password.go
package auth

import (
	"context"

	"golang.org/x/crypto/bcrypt"
)

const dummyPasswordHash = "$2a$10$7EqJtq98hPqEX7fNZaFWoOhiZCQGNuCIlnLzKeE/d1qFJKZ0f4uTe"

type PasswordVerifier struct{}

func (verifier PasswordVerifier) Verify(ctx context.Context, password string, passwordHash string) bool {
	_ = ctx
	return bcrypt.CompareHashAndPassword([]byte(passwordHash), []byte(password)) == nil
}

func (verifier PasswordVerifier) Hash(password string) (string, error) {
	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
	return string(hash), err
}

func (verifier PasswordVerifier) DummyHash() string {
	return dummyPasswordHash
}
